docs/contract-test-library.md

```diff
@@ -0,0 +1,541 @@
+---
+title: Taquito Smart Contract Library
+id: contracts_library
+author: Michael Kernaghan
+---
+
+The contracts used in Taquito Integration Tests and in Taquito Documentation Live Code Examples are test data and require curation. Here we collect the contracts, give them names, demonstrate their properties and describe their use.
+
+To determine if a contract has an FA1.2 interface we can use 
+```
+tezos-client check contract KT1CfFBaLoUrgv93k8668KCCcu2hNDNYPz4L implements fa1.2
+```
+A contract has an FA2 interface if it has entrypoints: transfer, balance_of, and update_operators
+- **Basic Contracts**
+  - [IncrementContract](#incrementcontract)
+  - [MichelsonMapContract](#michelsonmapcontract)
+- **Lambda Contracts**
+  - [LambdaViewContract](#lambdaviewcontract)
+  - [LambdaViewWithTokenContract](#lambdaviewwithtokencontract)
+- **Map and BigMap Contracts**
+  - [MapWithPairasMapContract](#mapwithpairasmapcontract)
+  - [MapWithValuesComplexKeysContract](#mapwithvaluescomplexkeyscontract)
+  - [MapWithInitialStorageContract](#mapwithinitialstoragecontract)
+  - [MapWithMapandBigmapContract](#mapwithmapandbigmapcontract)
+  - [BigMapsMultipleValuesContract](#bigmapsmultiplevaluescontract)
+  - [BigMapsComplexStorageContract](#smartcontractcomplexstoragecontract)
+  - [BigMapPackContract](#bigmappackcontract)
+- **Tzip7 Contracts**
+  - [TokenContract](#tokencontract)
+- **Tzip12 Contracts**
+  - [Tzip12BigMapOffChainContract](#tzip12bigmapoffchaincontract)
+- **Tzip 16 Contracts**
+  - [Tzip16StorageContract](#tzip16storagecontract)
+  - [Tzip16HTTPSContract](#tzip16httpscontract)
+  - [Tzip16SHA256Contract](#tzip16sha256contract)
+  - [Tzip16IPFSContract](#tzip16ipfscontract)
+  - [Tzip16OffChainContractOne](#tzip16offchaincontractone)
+  - [Tzip16OffChainContractTwo](#tzip16offchaincontracttwo)
+- **Wallet Contracts**
+  - [WalletContract](#walletcontract)
+  - [WalletAreYouThereContract](#walletareyouttherecontract)
+
+# Basic Contracts
+
+## IncrementContract
+
+[KT1NcdpzokZQY4sLmCBUwLnMHQCCQ6rRXYwS](https://better-call.dev/hangzhou2net/KT1NcdpzokZQY4sLmCBUwLnMHQCCQ6rRXYwS/code)
+
+This contract serves as the default introductory example on the [Ligo-WEB-IDE](https://ide.ligolang.org/p/CelcoaDRK5mLFDmr5rSWug)
+It has two endpoints, %decrement and %increment. The contract is used to demo addition and subtraction by a smart contract. This contract has neither an FA1.2 nor an FA2 interface.
+
+#### Entrypoints:
+  * decrement
+  * increment
+
+## MichelsonMapContract
+
+[KT1PAW3ghZyysrArcexyj6VUU7NZF8tHKoZR](https://better-call.dev/hangzhou2net/KT1PAW3ghZyysrArcexyj6VUU7NZF8tHKoZR/code)
+
+The contract supports a [Michelson Tutorial](https://tezostaquito.io/docs/michelsonmap). It has a default endpoint that takes a pair of an address and an amount of tez. 
+
+- [See the full tutorial](https://claudebarde.medium.com/?p=8d8be9930662)
+
+#### Entrypoints:
```

Comment:
Could it be valuable to have a section "storage" also for each contract that gives information on its type?

Reply:
i put a storage section for each contract and rearranged the ones I had already to be uniformly presented.